### DISCUSSION TITLE
Combining transactions in front-end

### DISCUSSION THREAD

**Discussion by hmzdot** [reach-sh/reach-lang#1041] [Q&A] [answered]
DevFracca's question
> Based on this response I'm assuming there's no way to get an optin prompt to occur from the Reach code in an atomc txn with a pay and delete? In the system I'm building the user has to first optin using  await this.reach.transfer(this.acc, this.acc, this.reach.bigNumberify(0), assetID); from the frontend, then they have to sign a separate txn for the pay and delete actions. 
>
> I'm getting some complaints that this process is very slow compared to the non smart contract solutions that other have built. Combining all of these txns into one would save a lot of time. 

```js
const ctcInfo = JSON.parse(chosenAppID);
      const ctc = this.acc.contract(backend, ctcInfo);
      const interact = {
        showBuyInfo: async (assetID, price) => {
          console.log(`AssetID: ${assetID} and Price: ${price}`);
          this.modalMsg = "Awaiting optin to asset..."
          await this.GetAssetURL(assetID);
          await this.reach.transfer(this.acc, this.acc, this.reach.bigNumberify(0), assetID); // optin
          this.modalMsg = "Awaiting payment and smart contract deletion..."
        }
      };
      await backend.Buyer(ctc, interact);
```

> Trying something like this which doesn't seem to be allowed in the rsh 
```js 
Buyer.pay(0, NFT).to(Buyer);
```

**Comment by jeapostrophe** [comment, marked as answer]
Correct, we do not currently have a way of exposing direct atomic group construction to frontends.
